src/pkg/services/m365/api/drive.go

```diff
@@ -266,19 +266,33 @@ func (c Drives) GetItemPermission(
 	ctx context.Context,
 	driveID, itemID string,
 ) (models.PermissionCollectionResponseable, error) {
-	perm, err := c.Stable.
-		Client().
-		Drives().
-		ByDriveId(driveID).
-		Items().
-		ByDriveItemId(itemID).
-		Permissions().
-		Get(ctx, nil)
-	if err != nil {
-		return nil, graph.Wrap(ctx, err, "getting item permission").With("item_id", itemID)
+	var (
+		retries int
+		perm    models.PermissionCollectionResponseable
+		err     error
+	)
+
+	// jwt retry has to be handled here, since we need to re-run the
+	// full client to get the new token.
+	for i := 0; i < 3; i++ {
```

Comment:
Could you please add a test for this? 
Side note, do we also need to worry about token expiry during item data download? That's separate from the download url jwt expiry, please correct me if wrong.

Reply:
That expiry should [already be handled](https://github.com/alcionai/corso/blob/main/src/internal/m365/collection/drive/collection.go#L345).  Seems like we had a gap in coverage here when it came to permissions downloads.

Reply:
Hmm, I think we're both seeing this at too small a scope.  It's probably something I should wrap into the client retry, rather than any one process that uses the client.  Let me go back and fix that.

Reply:
Yes, i was leading towards that :)

Reply:
done; final review is ready